### PR TITLE
Refactor span storage

### PR DESCRIPTION
**Pull request by savhappy**
[ WIP] Needs test adjustment and specs
  Considerations:
  Should we store root spans and child spans in separate tables?
  Storing key/values like this -> parent: `{{:root_span, span_id}, span}`  child: `{parent_span_id, span}`
  Should we implement some sort of sweep in case spans don't get removed `on_end`
  Should we order the child spans in the table? Or have an inserted_at value?

### CODE REVIEW

**Comment by savhappy** [comment]
@whatyouhide refactored with the genserver added. Left a comment wondering if we should sweep based on ttl or something else to avoid build up.

@solnic I have two failing test that I'm confused about.. 

``` elixir
  1) test sends captured spans as transactions with child spans (Sentry.Opentelemetry.SpanProcessorTest)
     test/sentry/opentelemetry/span_processor_test.exs:59
     ** (FunctionClauseError) no function clause matching in Calendar.ISO.parse_utc_datetime/2

     The following arguments were given to Calendar.ISO.parse_utc_datetime/2:

         # 1
         nil

         # 2
         :extended

     Attempted function clauses (showing 1 out of 1):

         def parse_utc_datetime(string, format) when is_binary(string) and (format === :basic or format === :extended)

     code: assert_valid_iso8601(transaction.timestamp)
     stacktrace:
       (elixir 1.17.2) lib/calendar/iso.ex:591: Calendar.ISO.parse_utc_datetime/2
       (elixir 1.17.2) lib/calendar/datetime.ex:1233: DateTime.from_iso8601/3
       test/sentry/opentelemetry/span_processor_test.exs:112: Sentry.Opentelemetry.SpanProcessorTest.assert_valid_iso8601/1
       test/sentry/opentelemetry/span_processor_test.exs:67: (test)
       ```
       
Thoughts?

File: lib/sentry/opentelemetry/span_storage.ex [hunk 1]
```diff
@@ -1,78 +1,81 @@
...
+    GenServer.start_link(__MODULE__, nil, name: name)
+  end
 
-      _ ->
-        :ok
-    end
+  @impl true
+  def init(nil) do
+    _table =
+      if :ets.whereis(@table) == :undefined do
```
Comment (whatyouhide) [inline lib/sentry/opentelemetry/span_storage.ex:16]: If the table is tied to this GenServer, where would we ever have the table running *without* it that we need this check?

File: lib/sentry/opentelemetry/span_storage.ex [hunk 2]
```diff
@@ -1,78 +1,81 @@
 defmodule Sentry.Opentelemetry.SpanStorage do
   @moduledoc false
+  use GenServer
 
-  @root_spans_table :sentry_root_spans
-  @child_spans_table :sentry_child_spans
+  @table :span_storage
```
Comment (whatyouhide) [inline lib/sentry/opentelemetry/span_storage.ex:5]: Scope this to Sentry, as the ETS namespace is global. My suggestion is

```suggestion
  @table __MODULE__
```

Comment (savhappy) [reply lib/sentry/opentelemetry/span_storage.ex:5]: In one of our pairing sessions for `check_in_id_mappings.ex`, we discussed this being bad practice to have both the GenServer and the table name the same. Thoughts?

**Comment by solnic** [comment]
> @solnic I have two failing test that I'm confused about..
> 
> ```elixir
>   1) test sends captured spans as transactions with child spans (Sentry.Opentelemetry.SpanProcessorTest)
>      test/sentry/opentelemetry/span_processor_test.exs:59
>      ** (FunctionClauseError) no function clause matching in Calendar.ISO.parse_utc_datetime/2
> 
>      The following arguments were given to Calendar.ISO.parse_utc_datetime/2:
> 
>          # 1
>          nil
> 
>          # 2
>          :extended
> 
>      Attempted function clauses (showing 1 out of 1):
> 
>          def parse_utc_datetime(string, format) when is_binary(string) and (format === :basic or format === :extended)
> 
>      code: assert_valid_iso8601(transaction.timestamp)
>      stacktrace:
>        (elixir 1.17.2) lib/calendar/iso.ex:591: Calendar.ISO.parse_utc_datetime/2
>        (elixir 1.17.2) lib/calendar/datetime.ex:1233: DateTime.from_iso8601/3
>        test/sentry/opentelemetry/span_processor_test.exs:112: Sentry.Opentelemetry.SpanProcessorTest.assert_valid_iso8601/1
>        test/sentry/opentelemetry/span_processor_test.exs:67: (test)
>      
> ```
> 
> Thoughts?

I'm looking into this finally. Sorry for late reply!

**Comment by solnic** [comment]
@savhappy fixed tests via https://github.com/getsentry/sentry-elixir/pull/817/commits/20534087c012d03ba9bec708221f2af722b35db4

@whatyouhide could you take another look please? would be good to merge this into the main PR branch and continue working there and wrap this up :)

I'm sorry that it took me so long to follow-up here!

**Pull request merged**